### ISSUE TITLE
Make c172p-canvas Default to Improve Frame Rate Performance

### ISSUE THREAD

**Issue by pedromorgan** [pedromorgan/flightgear-issues-test#1242]
_From [rogerxx@gmail.com](https://code.google.com/u/rogerxx@gmail.com/) on October 22, 2013 00:37:27_

After troubleshooting as to why I was getting some really poor framerates while using FlightGear 2.12.0 under Linux when compared to Windows, I have found a problem related to poor frame rates.  (Or at least with flightgear-2.12.0.)

It would appear the default "c172p" aircraft model loaded causes problems with poor frame rates.  Using the c172p-canvas solves the frame rate issues under Linux, and I'm now seeing frame rates comparable to Windows.

(Might also be wise to set default options similar to the Win32 build. ;-)

Can somebody point me to the ".c" source file where the default aircraft is specified, so I can provide a patch to the Gentoo ebuild files?  (Or, I can at least provide a notice after install if there is no simple method of augmenting the source code.)

_Original issue: http://code.google.com/p/flightgear-bugs/issues/detail?id=1243_

**Comment by pedromorgan** [comment]
_From [tom...@gmail.com](https://code.google.com/u/108827106934294482434/) on October 22, 2013 08:29:32_

Using the c172p-canvas is faster than using c172p?! They are nearly the same, only the c172p-canvas has an additional Canvas placed in the cockpit so I would not expect it to get faster. Are you sure that not something else has triggered this behavior?

**Status:** NeedInfo

**Comment by pedromorgan** [comment]
_From [rogerxx@gmail.com](https://code.google.com/u/rogerxx@gmail.com/) on October 22, 2013 12:42:12_

(Hey Tom.  I see you were CC'd concerning Bug `#1240` mentioned below.  Likely you're quite privy to what's going on here.)

I don't know.  I do know another major difference between Windows and Linux runs, the 'events' and  'xml-proprules' is quite excessive when flightgear is run under Linux when compared to Windows.

However, the increased 'events' and 'xml-proprules' (via Debug GUI monitor) is persistent and 'events' do not decrease when using the c172p-canvas aircraft model, while frame rates significantly increase!

Really odd.  I was more or less starting to focus on a bug with the lower level implementation of the simgear scripting, maybe also influenced by compile time options.  (I already see simgear has issues with -funroll-loops according to Bug `#1240` , and recompiling with certain flags also seems to affect frame rates.)  But this could also be due to the NVIDIA binary driver also going crazy, but I've seen absolutely no instability with the NVIDIA binary driver here on Linux.

About the only thing consistently reproduced at most times, is this option!  (Sporadically, once in a blue moon, the frame rates might dip to 25fps upon flightgear execution, and restarting flightgear or even restarting Linux as it maybe the NVIDIA binary driver, seems to solve the issue.)

If I had to point a finger, I'd point it at the simgear and the scripting.  I see nothing obvious the NVIDIA binary driver has anything to do with this.  Also, sporadic segfaults also seem to point to simgear including Issue `#1240` !

In other words, adding scripting might induce a delay within the execution of statements, inadvertently working around a bug.

I'll diff the aircraft models and see what I find.

Bottom line, 'c172p-canvas' model is rock steady 59-60 FPS while 'c172p' is consistently 25-30 FPS on Intel i7 here!

Both games-simulation/flightgear dev-games/simgear are compiled with stable flags per below.
\# Safe CFLAGS per Gentoo Wiki
CFLAGS="-O2 -march=corei7 -pipe"
CXXFLAGS="${CFLAGS}"

**Comment by pedromorgan** [comment]
_From [zakalawe@mac.com](https://code.google.com/u/zakalawe@mac.com/) on October 27, 2013 07:32:38_

Hmm, this makes very little sense to me either, the -canvas variant only adds some things, doesn't disable anything or replace it, so I am mystified how it could improve performance. Of course with graphics drivers nothing is impossible.

**Comment by pedromorgan** [comment]
_From [rogerxx@gmail.com](https://code.google.com/u/rogerxx@gmail.com/) on October 27, 2013 20:04:18_

I'm just reporting what I'm seeing with the FPS in game numbers, along with performance.

I've already found simgear doesn't work well with -funroll-loops.  (ie. no joystick/keyboard throttle)  I'll retest with my newly recompiled system.
